File: src/main/java/io/bindernews/thegrackle/cards/RocketGrackle.java
package io.bindernews.thegrackle.cards;

import com.megacrit.cardcrawl.actions.AbstractGameAction;
import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
import com.megacrit.cardcrawl.actions.common.DamageAction;
import com.megacrit.cardcrawl.cards.DamageInfo;
import com.megacrit.cardcrawl.core.AbstractCreature;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import com.megacrit.cardcrawl.powers.VulnerablePower;
import io.bindernews.bnsts.CardNums;
import lombok.val;

public class RocketGrackle extends BaseCard {
    public static final CardConfig C = new CardConfig("RocketGrackle", CardType.ATTACK);
    static final CardNums NUM = CardNums.builder()
            .cost(1)
            .damage(8).damageUpg(12)
            .magic(1).magicUpg(2)
            .build();

    public RocketGrackle() {
        super(C, CardRarity.COMMON, CardTarget.ENEMY);
        NUM.init(this);
        initializeDescription();
    }

    @Override
    public void apply(AbstractCreature p, AbstractCreature m) {
        val fx = AbstractGameAction.AttackEffect.BLUNT_HEAVY;
        val isMonster = p instanceof AbstractMonster;
        addToBot(new DamageAction(m, new DamageInfo(p, damage, damageTypeForTurn), fx));
        addToBot(new ApplyPowerAction(m, p, new VulnerablePower(m, magicNumber, isMonster), magicNumber));
    }

    @Override
    public void upgrade() {
        NUM.upgrade(this, false);
    }
}

File: src/main/kotlin/io/bindernews/thegrackle/cards/RocketGrackle.kt
package io.bindernews.thegrackle.cards

import com.megacrit.cardcrawl.actions.AbstractGameAction.AttackEffect
import com.megacrit.cardcrawl.actions.common.ApplyPowerAction
import com.megacrit.cardcrawl.actions.common.DamageAction
import com.megacrit.cardcrawl.cards.DamageInfo
import com.megacrit.cardcrawl.core.AbstractCreature
import com.megacrit.cardcrawl.monsters.AbstractMonster
import com.megacrit.cardcrawl.powers.VulnerablePower
import io.bindernews.bnsts.CardVariables

class RocketGrackle : BaseCard(C, VARS) {
    override fun apply(p: AbstractCreature, m: AbstractCreature?) {
        val fx = AttackEffect.BLUNT_HEAVY
        val isMonster = p is AbstractMonster
        addToBot(DamageAction(m, DamageInfo(p, damage, damageTypeForTurn), fx))
        addToBot(ApplyPowerAction(m, p, VulnerablePower(m, magicNumber, isMonster), magicNumber))
    }

    companion object {
        @JvmField val C = CardConfig("RocketGrackle", CardType.ATTACK, CardRarity.COMMON, CardTarget.ENEMY)
        val VARS = CardVariables.config { c ->
            c.cost(1, -1)
            c.damage(8, 12)
            c.magic(1, 2)
        }
    }
}